packages/boilerplate/src/reducers/page.js

```diff
@@ -1,11 +1,17 @@
-export default (state = 'HOME', action = {}) => components[action.type] || state
-
 const components = {
   HOME: 'Home',
   LIST: 'List',
   NOT_FOUND: 'NotFound',
 }
 
+export default (state = 'HOME', action = {}) => {
+  if (action.payload && action.payload.routes) {
+    Object.assign(components, action.payload.routes)
```

Comment:
This is a side effect in a reducer, which breaks the purity of redux, time travel, etc.

Perhaps you could implement something similar by having a separate reducer (or subkey in the existing rudy reducer) which stored references to the route components? You could dispatch actions in the constructor of the router/route components which registered themselves in this redux managed `componentMap`.

Is an app allowed multiple `<Router />`s like in `react-router`? in that case the contents would have to be scoped under some unique Id for each instance of `<Router />`